contracts/stellar-interchain-token-service/src/tests/register_canonical_token.rs

```diff
@@ -14,18 +14,22 @@ use crate::types::TokenManagerType;
 fn register_canonical_token_succeeds() {
     let (env, client, _, _, _) = setup_env();
     let token_address = Address::generate(&env);
-    let expected_id = client.canonical_interchain_token_id(&token_address);
+    let registered_token = &env.register_stellar_asset_contract_v2(token_address);
```

Comment:
Look carefully at what the input to this method should be, and not just whether it passes typecheck